test/integration/components/pingclient/pingclient.go

```diff
@@ -13,14 +15,30 @@ var tr = &http.Transport{
 var testHTTPClient = &http.Client{Transport: tr}
 
 func main() {
+	counter := 1
 	for {
-		r, err := testHTTPClient.Get("https://grafana.com")
+		req, err := http.NewRequest("GET", "https://grafana.com", nil)
+		if err != nil {
+			fmt.Println("error creating request:", err)
+			return
+		}
+		var traceID [16]byte
+		var spanID [8]byte
+		binary.BigEndian.PutUint64(traceID[:8], uint64(counter))
+		binary.BigEndian.PutUint64(spanID[:], uint64(counter))
+
+		// Generate a traceparent that we easily recognize
+		tp := fmt.Sprintf("00-%s-%s-01", hex.EncodeToString(traceID[:]), hex.EncodeToString(spanID[:]))
+		req.Header.Set("traceparent", tp)
```

Comment:
Would it be useful to have the option to not send the traceparent, so we can test both cases of with/without it?

Reply:
I didn't add one because I thought we already have a test for this, since we test the server spans for with or without traceID. It's the same code that generates both client and server spans, just different attributes. My main goal with this test was to ensure the bpf code captures the traceparent correctly on the client side, the rest is already well tested with the tests you added for the server side.